### ISSUE TITLE
Security and Privacy Properties of Groups

### ISSUE THREAD

**Issue by IamfromSpace** [simplex-chat/simplexmq#393]
I'm curious to understand just how it is that groups work, and what that means for their security and privacy properties.  For individual connections, this is all pretty well documented and clear.  For groups, it doesn't seem quite as clear to me exactly what compromises or concerns their may be.  It also isn't quite clear where groups are built on top of primitives, or if there are specific parts of the primitives that are indented for group behaviors.  It seems like the Open Connections RFC describes the current approach, and an "introduction" based approach was considered but eventually rejected?

**Comment by epoberezkin** [comment]
Replied to email - thank you!

TLDR:

Introducer can MITM the connection in the same way any service that participates in key exchange can MITM it (that is, any service other than SimpleX).

The group UI in the apps will both have the indication whether the connection is direct or via the introducer, and also allow to "secure the connection" by comparing key fingerprints via QR code scan.

**Comment by IamfromSpace** [comment]
I had a thought that Shamir Secret Sharing might be useful for a couple context—one being introductions.  If instead of sending your connection invite through one person, you split it among the parties, then an attack can only MITM if they control _every_ part (assuming fault tolerance isn’t used). This could be useful for allowing groups to securely introduce two parties.

It could serve as a way to make a connection over multiple otherwise insecure methods. This could just be called Multi-Factor Connections, where you can split the invite multiple ways.  You can then send each portion via different channel of your choosing: SMS, email, group chats, snail mail, other SimpleX users, etc.  Once all collected, they can be reconstructed into the single invite.

Happens to also simplify my groups RFC, which is the benefit in figuring that out over TLA+ rather than shipped code :)

Multi-Factor Invites seem pretty compelling to me though, I’m curious if that take is shared.

**Comment by epoberezkin** [comment]
It achieves the same as validating the connection after the fact (via another channel), right?

**Comment by IamfromSpace** [comment]
Oh, but yes, I see, confirming invite details or connection fingerprints over a secondary channel should do more-or-less the same thing.

I think the appeal in this case is the validation is essentially automatic.  You use and confirm every channel you use.
